extension/main.js

```diff
@@ -16,13 +16,31 @@
 	}
 
 	function update() {
-		gitHubNotifCount(function (count) {
+		gitHubNotifCount(function (count, interval) {
+			var period = 1;
+			if (interval !== GitHubNotify.settings.get('interval')) {
+				GitHubNotify.settings.set('interval', interval);
+				period = Math.ceil(interval / 60);
+				if (period < 1) {
+					period = 1;
+				}
+				chrome.alarms.clearAll(function () {
+					chrome.alarms.create({periodInMinutes: period});
+					chrome.alarms.onAlarm.addListener(update);
+					chrome.runtime.onMessage.addListener(update);
+				});
+			}
 			if (count < 0) {
 				var text;
 				if (count === -1) {
-					text = 'You have to be connected to the internet and logged into GitHub';
+					text = 'You have to be connected to the internet';
 				} else if (count === -2) {
-					text = 'Unable to find count on page';
+					text = 'Unable to find count';
+				} else if (count === -3) {
+					// not-modified, do not re-render
+					return;
+				} else if (count === -4) {
+					text = 'Missing access token, please create one and enter it in Options';
 				}
 				render('?', [166, 41, 41, 255], text);
 			} else {
```

Comment:
Scratch this one as well. See my below comment why this is a string in the first place.